### PR TITLE
Redirect and resend verification email when pending user attempts to login

### PR DESCRIPTION
**Pull request by imdadahad**
When a pending user attempts to create an account/login they will receive another verification email (incase the email was deleted/in spam)

@quis does the message on line 41 sound okay, can amend if required?

### CODE REVIEW

File: app/main/views/sign_in.py [hunk 1]
```diff
@@ -38,8 +38,8 @@ def sign_in():
         user = user_api_client.get_user_by_email_or_none(form.email_address.data)
         user = _get_and_verify_user(user, form.password.data)
         if user and user.state == 'pending':
-            flash("You haven't verified your email or mobile number yet.")
-            return redirect(url_for('main.sign_in'))
+            flash("You haven't verified your email or mobile number yet. Check your email for a verification link.")
```
Comment (quis) [inline app/main/views/sign_in.py]: I don’t think you need the flash message at all here. The page will look like:

>   **Check your email**
>
>   In order to verify your email address we've sent a new confirmation link to test@example.com

Comment (quis) [inline app/main/views/sign_in.py]: And I’ve tweaked it a bit here: https://github.com/alphagov/notifications-admin/pull/927

Comment (imdadahad) [inline app/main/views/sign_in.py]: Gotcha, just updated.

**Comment by quis** [comment]
Tested locally, looks good 👍

**Pull request merged**